app/frontend/src/cljs/teet/project/land_view.cljs

```diff
@@ -907,73 +905,87 @@
+  [:div {:class (<class project-style/owner-info)}
+   [:div {:style {:display :flex
+                  :flex-direction :row
+                  :justify-content :space-between}}
+    [:div {:style {:margin-bottom "0.5rem"}}
+     [typography/BoldGreyText (if person?
+                                (str eesnimi " " nimi)
+                                nimi)]
+     [typography/GreyText (str (tr [:land :owner-code]) " " r_kood)]]
+    [:div
+     (when omandiosa_suurus
+       [typography/BoldGreyText
+        (when first-in-joint?
+          (if (= "1" omandiosa_lugeja omandiosa_nimetaja)
+            "1"
+            (str omandiosa_lugeja "/" omandiosa_nimetaja)))])]]
+   (when (and (= isiku_tyyp "Juriidiline isik") r_kood) ;; r_kood was null in some cases in production data
+     [:div {:style {:padding-left "0.5rem"
+                    :margin-bottom "0.5rem"
+                    :border-left (str "solid 7px " theme-colors/gray-light)}}
+      [query/query {:e! e!
+                    :query :land/estate-owner-info
+                    :args {:thk.project/id (:thk.project/id project)
+                           :business-id r_kood}
+                    :simple-view [business-registry-info]}]])])
+
+(defn- owner-is-person? [{:keys [isiku_tyyp]}]
+  (= isiku_tyyp  "Füüsiline isik"))
 
 (defn one-owner-for-modal
-  [estate-info e! app project
-   {:keys [omandiosa_suurus omandiosa_lugeja isik omandiosa_nimetaja] :as owner}]
-  (let [isiku_tyyp (:isiku_tyyp (first isik))
-        person? (= isiku_tyyp  "Füüsiline isik")
-        business-ids (map :r_kood
-                          (filter #(not= (:isiku_tyyp %) "Füüsiline isik")
-                                  (mapcat :isik owner)))
-        joint-ownership? (> (count isik) 1)]
-    ;; (log/debug "isiku_tyyp =" isiku_tyyp ", joint-ownership?" joint-ownership?)
-    [:<>
-     (for [[i {:keys [r_kood eesnimi nimi]}] (map-indexed vector isik)]
-       
-       [owner-inner-component e! person? nimi eesnimi r_kood omandiosa_lugeja omandiosa_nimetaja omandiosa_suurus isiku_tyyp project (= i 0)])
-     (for [r_kood business-ids]
-       [comments-view/lazy-comments {:e! e!
-                                     :app app
-                                     :entity-type :owner-comments
-                                     :entity-id [:owner-comments/project+owner-id [(:db/id project) r_kood]]}])]))
+  [_estate-info e! app project
+   {:keys [omandiosa_suurus omandiosa_lugeja isik omandiosa_nimetaja]}]
+  [:div
+   [:div {:class (<class project-style/owner-container)}
+    (for [[i {:keys [r_kood eesnimi nimi isiku_tyyp] :as owner}] (map-indexed vector isik)]
+      [:<>
+       [owner-inner-component e! (owner-is-person? owner) nimi eesnimi r_kood omandiosa_lugeja omandiosa_nimetaja omandiosa_suurus isiku_tyyp project (= i 0)]])]
+   ;; Here we're making two assumptions:
+   ;; 1) Comments are shown only for non-person owners (ie. companies and such)
```

Comment:
It occurred to me just now that IIRC business/person id can be null/empty also in case of state owned land. This might also be the case to which another comment of observed null r_kood's refs to.